### ISSUE TITLE
Poor Eval performance with Nerfacto with default poster data

### ISSUE THREAD

**Issue by kudo1026** [nerfstudio-project/nerfstudio#1100]
**Describe the bug**
The eval performance on the default poster dataset with the Nerfacto method is not ideal.

**To Reproduce**
1. Simply run the most basic command `ns-train nerfacto` with any vis you like (I use tensorboard here).

**Expected behavior**
With the default method and default dataset, I expect it to get both good visualization and good metric numbers. However, while the training PSNR does look good (~32-33) and the visualization does "look" good, the actual eval PSNR number is not good. It starts around 24, and oscillate between 16 and 26, but never reaches 30+, which is usually roughly considered the bar for a "good" reconstruction. 

**Screenshots**
![image](https://user-images.githubusercontent.com/16885585/206234767-2a601878-0df6-44ae-ab17-9229f3014f67.png)
As shown from the above screenshots, the eval PSNR never gets good.
![image](https://user-images.githubusercontent.com/16885585/206239587-6809f154-8d9b-427c-bbc1-248e3396e571.png)
If you compare the rendered eval images side-by-side with gt, it seems that while the shapes are captured pretty well, the actual color rendered is a bit off, which contribute to the poor eval PSNR that is directly calculated on the pixel values.

**Additional context**
As a side note, the instant-ngp method included in the repo actually performs well with none of the problems mentioned above so I could always opt to use that if anything. However, it does feel weird that the default method with so many good tricks implemented can still have this undesirable behavior. I'm interested to know what you think could be the reasons for it.

**Comment by tancik** [comment]
There are likely two causes for the lower psnr, pose optimization and appearance matching.

Pose: During the training we jointly optimize the poses as the input poses might have a bit of noise. Since this is only done to the training poses, it is possible that the eval poses will no longer match up as well leading to a lower PSNR. To get around this there are a few solutions, disable pose optimization `--pipeline.datamanager.camera-optimizer.mode off`. This may result in visually worse reconstructions, but better PSNR (we default to keeping it on since we are optimizing for best visual quality). Alternatively you could keep the pose optimization on, but run pose optimization on the eval images before testing for PSNR. This provides best of both worlds but makes the eval more complex. We are considering adding this feature.

Appearance Matching: In real-world captures it is very likely that the appearance changes between images (different exposures, white balance, ect). To account for this we use GLO appearance embeddings to allow the network to fine-tune to each training examples appearance. This helps significantly with "floaters" and other artifacts that are more prevalent in methods like the original nerf, or instant-ngp. However, similar to the pose optimization, we are left with the issue of what "appearance" do we use when evaluating the PSNR, or rendering any new view? In nerfacto we opt to use the mean appearance. In the example you provided, it is clear that the "mean appearance" is much brighter than that particular eval example. So similar to pose optimization, you can get better PSNR by disabling appearance embeddings, however the visual quality will degrade and more floaters will be present. A workaround is to optimize the appearance on half of the images and only calculate the PSNR on the other half. This is what some papers like NeRF-W do. We may add these evaluations in the future, but they are a bit more complex and only work well when computing imagewise metrics, not pixelwise metrics.
